### PR TITLE
Integrate data pages

### PR DESCRIPTION
**Pull request by churtado-edrans**
## Description

This PR aims to define a new Data resource to retrieve the PAGE ID of a specific page name.

### CODE REVIEW

File: examples/data-sources/statuspage_pages/data-source.tf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+data "statuspage_pages" "default" {
+  page_id = local.page_name
```
Comment (sbecker59) [inline examples/data-sources/statuspage_pages/data-source.tf:2]: If I look at the schema in the [data_sources_page ](https://github.com/edrans/terraform-provider-statuspage/blob/integrate-data-pages/statuspage/data_source_pages.go) file, shouldn't I replace `page_id `with `page_name` ?

File: statuspage/data_source_pages.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+		Schema: map[string]*schema.Schema{
+			"filter": dataSourceFiltersSchema(),
+			"page_name": {
+				Description:  "the name of the page to",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			// Computed values
```
Comment (sbecker59) [inline statuspage/data_source_pages.go:21]: Comment code is not helpful

File: statuspage/data_source_pages.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+func dataSourcePagesRead(d *schema.ResourceData, m interface{}) error {
+
+	providerConf := m.(*ProviderConfiguration)
+	statuspageClientV1 := providerConf.StatuspageClientV1
+	authV1 := providerConf.AuthV1
+	page_name := d.Get("page_name").(string)
+
+	req := statuspageClientV1.PagesApi.GetPages(authV1)
```
Comment (sbecker59) [inline statuspage/data_source_pages.go:55]: I will replace the 2 lines below for better error management
```go
req, _, err := statuspageClientV1.PagesApi.GetPages(authV1).Execute()
if err.Error() != "" {
     return TranslateClientErrorDiag(err, "error querying pages list")
}

```
What do you think ?

**Review by sbecker59 — CHANGES_REQUESTED** [review]
Hi @churtado-edrans 
Thanks for the contribution :tada:

:test_tube: Can you create a test on this new datasource (e.g. [data_source_component_groups_test](https://github.com/sbecker59/terraform-provider-statuspage/blob/main/statuspage/data_source_component_groups_test.go) please?
:wrench: For information, for the unit test my test page is called `Huskyly`


:memo: Can you generate the documentation for the terraform registry from the example using [tfplugindocs](https://www.terraform.io/registry/providers/docs)

:bulb: I put some comments on the current proposal

:busts_in_silhouette: If you have any questions or need help, don't hesitate.

File: statuspage/data_source_pages.go (hunk 3 above)
Comment (churtado-edrans) [reply statuspage/data_source_pages.go:55]: When err is managed in code the method returns the following error.

`error(*encoding/json.UnmarshalTypeError) *{Value: "object", Type: reflect.Type(*reflect.rtype) *{size: 16, ptrdata: 8, hash: 3774831796, tflag: tflagUncommon|tflagExtraStar|tflagNamed (7), align: 8, fieldAlign: 8, kind: 24, equal: runtime.strequal, gcdata: *1, str: 15712, ptrToThis: 319616}, Offset: 1104, Struct: "Page", Field: "favicon_logo"}`

This is the reason I avoid using it temporarily to handle the error. I tried to follow up on your status page client but i don't find the reason.

Can you help me with that?

File: statuspage/data_source_pages.go (hunk 2 above)
Comment (churtado-edrans) [reply statuspage/data_source_pages.go]: Removed comments unused

**Comment by churtado-edrans** [comment]
I added a new mandatory  STATUSPAGE_PAGE_NAME variable to make data pages test. Can you add in your environment this variable name?

File: statuspage/data_source_pages.go (hunk 3 above)
Comment (sbecker59) [reply statuspage/data_source_pages.go:32]: I updated the statuspage-api-client-go client version. 
The new [version ](https://github.com/sbecker59/statuspage-api-client-go/commit/7305a2c56c57de1d945bb7b2ce3fe2ee193d5d06#diff-38e50faa0b004f9f015df27cbc0949a192fe7ceb72e99a933cfae6a4ddd22a97)is `github.com/sbecker59/statuspage-api-client-go v0.0.0-20220315194540-7305a2c56c57`

There is a problem between the swagger provided by the statuspage api and the actual result.
To generate the `statuspage-api-client-go` I used openapi-generator which matches the [swagger file](https://developer.statuspage.io/) as a go client


Type in swagger:
```go
FaviconLogo *string `json:"favicon_logo,omitempty"`
```

Type in result: 
```json
"favicon_logo": {
   "updated_at": null,
   "size": null,
   "urls": ""
}
```

**Comment by sbecker59** [comment]
I'm going to merge this PR and update the dependencies in a following PR.

**Comment by sbecker59** [comment]
The new version of the provider 0.8.0 included now the management of the [statuspage_pages](https://registry.terraform.io/providers/sbecker59/statuspage/latest/docs/data-sources/pages).

**Comment by churtado-edrans** [comment]
Thanks for your help and for upgrading the package! It's a pleasure!

**Pull request merged**